wip/services/exec/store.go

```diff
@@ -18,55 +17,67 @@ func NewPostgresStore() Store {
 	return postgresStore{}
 }
 
-func (p postgresStore) Create(project string, exec types.Exec) error {
-	var command []string
-	var commitID, gitRemoteURL sql.NullString
+func (p postgresStore) Create(projectID string, exec types.Exec) error {
+	ctx := context.Background()
 
-	if exec.Command != nil {
-		command = exec.Command
-	}
-	if exec.CommitID != nil {
-		commitID = sql.NullString{String: *exec.CommitID, Valid: true}
-	}
-	if exec.GitURL != nil {
-		gitRemoteURL = sql.NullString{String: *exec.GitURL, Valid: true}
+	if projectID == "" {
+		return fmt.Errorf("an Exec must be attached to a project")
 	}
-	bid := sql.NullString{}
-	if exec.BuildID != nil {
-		bid = sql.NullString{String: *exec.BuildID, Valid: true}
+
+	// Every exec should be created with a public key
+	publicKeys := filterNullPublicKeys(exec.Keys)
+	if len(publicKeys) > 1 || len(publicKeys) == 0 {
+		return fmt.Errorf("an Exec must be created with one and only one SSH public key")
 	}
 	if exec.Name == "" {
-		exec.Name = random.GenerateRandomPhrase(4, "-")
+		return fmt.Errorf("an Exec must be named")
```

Comment:
This is not right. We don't want the user to name an exec. We give it a human readable name as a proxy for an id unique to a project